terraform/api_gateway.tf.json.template.py

```diff
@@ -504,7 +556,157 @@ def check_waf_rules(rules: JSONs) -> JSONs:
+                                                                'positional_constraint': 'EXACTLY',
+                                                                'search_string': 'PUT',
+                                                                'text_transformation': {
+                                                                    'priority': 0,
+                                                                    'type': 'NONE'
+                                                                }
+                                                            }
+                                                        },
+                                                        {
+                                                            'regex_match_statement': {
+                                                                'regex_string': '^(/fetch)?/manifest/files',
+                                                                'field_to_match': {
+                                                                    'uri_path': {}
+                                                                },
+                                                                'text_transformation': {
+                                                                    'priority': 0,
+                                                                    'type': 'NONE'
+                                                                }
+                                                            }
+                                                        }
+                                                    ]
+                                                }
+                                            ]
+                                        }
+                                    }
+                                },
+                                'action': {
+                                    'block': {
+                                        'custom_response': {
+                                            'response_code': 429,
+                                            'response_header': [
+                                                {
+                                                    'name': 'Retry-After',
+                                                    'value': str(config.waf_rate_limit_manifests.retry_after)
+                                                }
+                                            ]
+                                        }
+                                    }
+                                },
+                                'visibility_config': {
+                                    'metric_name': config.waf_rate_limit_manifests.name,
+                                    'sampled_requests_enabled': True,
+                                    'cloudwatch_metrics_enabled': True
+                                }
+                            }
```

Comment:
Deduplicate. It's OK to add explicit condition on GET for the (/fetch)/repository/files endpoint.